ios/RNGoogleMobileAds/RNGoogleMobileAdsNativeView.mm

```diff
@@ -0,0 +1,179 @@
+
+#import "RCTFabricComponentsPlugins.h"
+#endif
+
+@interface RCTBridge (Private)
++ (RCTBridge *)currentBridge;
+@end
+
+#ifdef RCT_NEW_ARCH_ENABLED
+using namespace facebook::react;
+
+@interface RNGoogleMobileAdsNativeView () <RCTRNGoogleMobileAdsNativeViewViewProtocol>
+@end
+#endif
+
+@implementation RNGoogleMobileAdsNativeView {
+  __weak RCTBridge *_bridge;
+  __weak RNGoogleMobileAdsNativeModule *_nativeModule;
+  __weak GADNativeAd *_nativeAd;
+  GADNativeAdView *_nativeAdView;
+  dispatch_block_t _debouncedReload;
+}
+
+#ifdef RCT_NEW_ARCH_ENABLED
+#pragma mark - Fabric specific
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  if (self = [super initWithFrame:frame]) {
+    static const auto defaultProps = std::make_shared<const RNGoogleMobileAdsBannerViewProps>();
+    _props = defaultProps;
+
+    _bridge = [RCTBridge currentBridge];
+    _nativeModule = [_bridge moduleForClass:RNGoogleMobileAdsNativeModule.class];
+    _nativeAdView = [[GADNativeAdView alloc] init];
+    self.contentView = _nativeAdView;
+  }
+
+  return self;
+}
+
+#pragma mark - RCTComponentViewProtocol
+
++ (ComponentDescriptorProvider)componentDescriptorProvider {
+  return concreteComponentDescriptorProvider<RNGoogleMobileAdsNativeViewComponentDescriptor>();
+}
+
++ (BOOL)shouldBeRecycled {
+  return NO;
+}
+
+- (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps {
+  const auto &oldViewProps =
+      *std::static_pointer_cast<RNGoogleMobileAdsNativeViewProps const>(_props);
+  const auto &newViewProps =
+      *std::static_pointer_cast<RNGoogleMobileAdsNativeViewProps const>(props);
+
+  if (oldViewProps.responseId != newViewProps.responseId) {
+    NSString *responseId = [[NSString alloc] initWithUTF8String:newViewProps.responseId.c_str()];
+    _nativeAd = [_nativeModule nativeAdForResponseId:responseId];
+    [self reloadAd];
+  }
+
+  [super updateProps:props oldProps:oldProps];
+}
+
+- (void)handleCommand:(const NSString *)commandName args:(const NSArray *)args {
+  RCTRNGoogleMobileAdsNativeViewHandleCommand(self, commandName, args);
+}
+
+#else
+#pragma mark - Paper specific
+
+- (instancetype)initWithBridge:(RCTBridge *)bridge {
+  if (self = [super init]) {
+    _bridge = bridge;
+    _nativeModule = [_bridge moduleForClass:RNGoogleMobileAdsNativeModule.class];
+    _nativeAdView = self;
+  }
+  return self;
+}
+
+#endif  // RCT_NEW_ARCH_ENABLED
+
+#pragma mark - Common logics
+
+- (void)registerAsset:(NSString *)assetKey reactTag:(NSInteger)reactTag {
+  RCTExecuteOnMainQueue(^{
+    UIView *view = [_bridge.uiManager viewForReactTag:@(reactTag)];
+
+    if ([assetKey isEqual:@"media"] && [view isKindOfClass:RNGoogleMobileAdsMediaView.class]) {
+#ifdef RCT_NEW_ARCH_ENABLED
+      GADMediaView *mediaView = ((RNGoogleMobileAdsMediaView *)view).contentView;
+#else
+      GADMediaView *mediaView = (RNGoogleMobileAdsMediaView *) view);
```

Comment:
Is inserted an extra parenthesis.
It should be `GADMediaView *mediaView = (RNGoogleMobileAdsMediaView *)view;`